include/zephyr/bluetooth/services/ots.h

```diff
@@ -902,6 +902,17 @@ struct bt_ots_client_cb {
  */
 int bt_ots_client_register(struct bt_ots_client *ots_inst);
 
+/** @brief Unregister an Object Transfer Service Instance.
+ *
+ *  Unregister an Object Transfer Service instance discovered on the peer.
+ *  Call this function when the BLE peer disconnected.
+ *
+ *  @param[in]  index      Discovered OTS instance.
+ *
+ *  @return int            0 if success, ERRNO on failure.
+ */
+int bt_ots_client_unregister(uint8_t index);
```

Comment:
There is no way for the user of this API to get the index. I would prefer `bt_ots_client_unregister(struct bt_ots_client *ots_inst)`. Alternatively, change `bt_ots_client_register` so that non-negative return values are the assigned index. Or add a `bt_ots_get_client_handle`/`index`.

Reply:
Currently the OTS client instance number is hardcoded as 1. So user should just assign 0 to the index.
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/services/ots/ots_client.c#L35
I have no idea if the original maintainer plan to do multi instances implementation.

Reply:
> Currently the OTS client instance number is hardcoded as 1. So user should just assign 0 to the index. https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/services/ots/ots_client.c#L35 I have no idea if the original maintainer plan to do multi instances implementation.

The API and the current implementation are different things. We have to future-proof the API.

The `index`-approach does not make sense in this API. We can document that `index` must always be `0`, and other values are RFU. But that feels unwise to me because we are heading down a path where the user must keep track of the indexes. And the indexes should not be needed, since the `ots_inst`s are unique (right?). It smells to me of a leaked implementation detail - that the implementation is storing `ots_inst` pointers in an array.

I still suggest changing the prototype to `bt_ots_client_unregister(struct bt_ots_client *ots_inst)`. Another possibility is to have a `bt_ots_client_unregister_all(void)` instead.